### PR TITLE
Add deflate and inflate of panel content, changed gc chance from 1% t…

### PR DESCRIPTION
**Pull request by Harfusha**
…o 10%, added 10% chance to vacuum when sqlite is used, added pragma commands for optimalization when using sqlite

https://github.com/cakephp/debug_kit/issues/1021

### CODE REVIEW

File: src/Model/Entity/Panel.php [hunk 1]
```diff
@@ -46,10 +46,37 @@ class Panel extends Entity
      */
     protected function _getContent(mixed $content): string
     {
+        if (is_string($content)) {
+            $contentInflated = @gzinflate($content);
+            if ($contentInflated) {
```
Comment (markstory) [inline src/Model/Entity/Panel.php]:
```suggestion
            if ($contentInflated !== false) {
```

comparing to false will avoid empty string from being ignored.

File: src/Model/Entity/Panel.php [hunk 2]
```diff
@@ -46,10 +46,37 @@ class Panel extends Entity
      */
     protected function _getContent(mixed $content): string
     {
+        if (is_string($content)) {
+            $contentInflated = @gzinflate($content);
```
Comment (LordSimal) [inline src/Model/Entity/Panel.php:55]: According to https://www.php.net/manual/en/zlib.installation.php zlib is not installed by default via PHP.

To not force all users to install the php zlib extension I'd simply wrap that in a 
```
if(function_exists('gzinflate'))
``` 
to make sure this is actually present.

File: src/Model/Entity/Panel.php [hunk 3]
```diff
@@ -46,10 +46,37 @@ class Panel extends Entity
...
+     * Deflate the string data before saving it into database
+     *
+     * @param mixed $content Content
+     * @return mixed
+     */
+    protected function _setContent(mixed $content): mixed
+    {
+        if (is_string($content)) {
+            $contentDeflated = gzdeflate($content, 9);
+            if ($contentDeflated) {
```
Comment (markstory) [inline src/Model/Entity/Panel.php]:
```suggestion
            if ($contentDeflated !== false) {
```

Same here.

File: src/Model/Table/RequestsTable.php [hunk 4]
```diff
@@ -131,7 +141,26 @@ public function gc(): void
...
             if ($conn->getDriver() instanceof Sqlite) {
-                $conn->execute('VACUUM;');
+                $conn->execute('
+                    PRAGMA auto_vacuum = FULL;
+                    PRAGMA journal_mode = OFF;
+                    PRAGMA synchronous = OFF;
+                    PRAGMA foreign_keys = OFF;
+                    PRAGMA temp_store = MEMORY;
+                    PRAGMA automatic_index = OFF;
+                ');
```
Comment (markstory) [inline src/Model/Table/RequestsTable.php:152]: Should this only happen if the GC is going to occur?

File: src/Model/Table/RequestsTable.php [hunk 5]
```diff
@@ -131,7 +141,26 @@ public function gc(): void
...
+
+                if (!$this->shouldGcVacuum()) {
```
Comment (markstory) [inline src/Model/Table/RequestsTable.php:154]: If we've already passed the rand to do a GC why do we need another rand to do vacuum?

File: src/Model/Table/RequestsTable.php (hunk 4 above)
Comment (Harfusha) [reply src/Model/Table/RequestsTable.php:152]: I think it is not needed to run everytime, so it is run when gc is ran

File: src/Model/Table/RequestsTable.php (hunk 5 above)
Comment (Harfusha) [reply src/Model/Table/RequestsTable.php:154]: So rows are cleared more frequently, but the slow vacuum should be ran only 1% of requests (I made the check to own functions for static analysis tools and if we want to tweak the chance)

**Comment by Harfusha** [comment]
Any other issues with this?

**Pull request merged**